src/SIL.XForge.Scripture/Services/ParatextSyncRunner.cs

```diff
@@ -741,17 +741,38 @@ CancellationToken token
 
     internal void CloseConnection() => _conn?.Dispose();
 
-    private async Task UpdateParatextBook(
+    /// <summary>
+    /// Returns an XDocument for the USX of a book in a Paratext project repo.
+    /// </summary>
+    internal XDocument GetBookUsx(string ptProjectId, int bookNum)
```

Comment:
I noticed a bug on the first sync after switching branches to this PR, where code that was affected by this bug had the space removed in Paratext when I synced in PT.

For example, this line of USFM:

```
\q1
\v 4 He sent his messenger\f + \fr 1.4 \fq messenger \ft Or \fqa angel\f* \add to me\add*, took me from my father’s sheep,
\q2 and anointed me with olive oil.\x - \xo 1.4 \xt 1 Sam 16:13\x*
```

Rendered (incorrectly) in Scripture Forge as:

![image](https://user-images.githubusercontent.com/8665431/229390008-b48ce842-fe7f-4e4d-8798-6b9a78a0952a.png)

After switching to this branch, syncing in Scripture Forge, then syncing in Paratext, the USFM in Paratext was changed to:

```
\q1
\v 4 He sent his messenger\f + \fr 1.4 \fq messenger \ft Or \fqa angel\f*\add to me\add*, took me from my father’s sheep,
\q2 and anointed me with olive oil.\x - \xo 1.4 \xt 1 Sam 16:13\x*
```

And still displayed like this in Scripture Forge:

![image](https://user-images.githubusercontent.com/8665431/229390079-f0221e6d-df39-4256-9b90-1da4cf248eb1.png)

Adding the space in PT then syncing obviously fixed this (both in PT then in SF after syncing in SF), but I only noticed this bug because I trawled through the Mercurial commits.

I am not sure on a fix for this bug, as it only occurs once, immediately after this change is deployed (future syncs used your fix and worked perfectly).

Thinking out loud....one way could be to store a version number with the project keeping track of which parser/sync version was used to sync the project (maybe in sf_project, default to 1), and when a sync is run after the parser/sync is updated to a new version, the old v1 USX parser would be called, then right before we load the new USFM from Paratext, we would increment the version in the sf_project, then use the new parser from here on. Seems kind of messy - you might have a cleaner solution?

Reply:
I'm a bit confused by the series of events. Are you saying:
1. You have a Paratext project with USFM which contains text that has a space.
2. You use the current SF master branch and you have SF sync. SF imported the text but omitted the space.
3. You switch to using this PR branch.
4. You have SF sync. SF exports the lost-space version of the text.
5. In PT you do a SendReceive. It now has the lost-space version of the text.

> I am not sure on a fix for this bug, as it only occurs once, immediately after this change is deployed (future syncs used your fix and worked perfectly).

If the above is what you're meaning, step 3 isn't necessary for step 4 to propagate the lost-space version of the text to PT. Once the lost-space version of the text is in SF (from step 2), it's going to flow back out whether you stay on master branch or first switch to the PR branch. But maybe I'm not understanding what you're saying.


Any project text that was imported already could have lost some spaces between styled regions. I suppose we could now try to detect instances where we are removing a space between two styles regions when writing from SF to PT, and ... maybe do something clever so we don't write back bad changes. If we can be sure which are actually bad vs intentional.

Reply:
(Okay, so the problem is:
When we import a project into SF, SF can lose some spaces. When we write that back to PT, we propagate the loss of the spaces. 
But we won't write back to PT and propagate the loss _unless the chapter is edited/changed_. 
Back when we were losing spaces, we lost spaces on both sides of the change comparison, and so an unchanged chapter did not write the space-loss back to PT because we did not think the chapter was changed. But now that we don't lose the spaces, when building data for comparison, if there _were_ spaces lost, as represented in the current chapter doc deltas, then that chapter will be flagged as _changed_ (because it looks like the space was deleted in SF). And so we will write the space-loss out to PT. )